File: js/runtime/bridge.js
const IS_TAURI = '__TAURI__' in window;

async function handle(action, target) {
    if (!action) return;

    if (IS_TAURI) {
        const { invoke } = window.__TAURI__.core;
        await invoke('native_action', { action, target: target || '' });
    } else {
        handleWeb(action, target);
    }
}

function handleWeb(action, target) {
    switch (action) {
        case 'so:alert':
            alert(target);
            break;
        case 'so:notification':
            if ('Notification' in window && Notification.permission === 'granted') {
                new Notification('NativeBlade', { body: target });
            } else {
                alert(target);
            }
            break;
        case 'so:navigate':
            window.location.href = target;
            break;
        case 'so:exit':
            window.close();
            break;
    }
}

export default { handle, IS_TAURI };

File: rust/src/commands/bridge.rs
use tauri_plugin_dialog::DialogExt;
use tauri_plugin_notification::NotificationExt;

#[tauri::command]
pub fn native_action(app: tauri::AppHandle, action: String, target: String) {
    match action.as_str() {
        "so:alert" => {
            app.dialog().message(&target).title("NativeBlade").blocking_show();
        }
        "so:notification" => {
            let _ = app.notification().builder().title("NativeBlade").body(&target).show();
        }
        "so:exit" => {
            app.exit(0);
        }
        _ => {}
    }
}
